Sources/TraceLog/Configuration.swift

```diff
@@ -0,0 +1,177 @@
+            
+            if upperCaseVariable.hasPrefix(Configuration.logAll) {
+                
+                if let level = upperCaselogLevelString.asLogLevel() {
+                    self.globalLogLevel = level
+                } else {
+                    errors.append(.invalidLogLevel("Variable '\(upperCaseVariable)' has an invalid logLevel of '\(upperCaselogLevelString)'. '\(Configuration.logAll)' will be set to \(self.globalLogLevel)."))
+                }
+                
+            } else if upperCaseVariable.hasPrefix(Configuration.logPrefix) {
+                
+                if let level = upperCaselogLevelString.asLogLevel() {
+                    ///
+                    /// Note: in order to allow for case sensitive tag prefix names, we use the variable instead of the uppercase version.
+                    ///
+                    if let logLevelScopeRange = variable.range(of: Configuration.logPrefix) {
+                        let logLevelScope = variable.substring(from: logLevelScopeRange.upperBound)
+                        
+                        self.loggedPrefixes[logLevelScope] =  level
+                    }
+                } else {
+                    errors.append(.invalidLogLevel("Variable '\(upperCaseVariable)' has an invalid logLevel of '\(upperCaselogLevelString)'. '\(upperCaseVariable)' will NOT be set."))
+                }
+                
+            } else if upperCaseVariable.hasPrefix(Configuration.logTag) {
+                
+                if let level = upperCaselogLevelString.asLogLevel() {
+                    ///
+                    /// Note: in order to allow for case sensitive tag names, we use the variable instead of the uppercase version.
+                    ///
+                    if let logLevelScopeRange = variable.range(of: Configuration.logTag) {
+                        let logLevelScope      = variable.substring(from: logLevelScopeRange.upperBound)
+                        
+                        self.loggedTags[logLevelScope] =  level
+                    }
+                } else {
+                    errors.append(.invalidLogLevel("Variable '\(upperCaseVariable)' has an invalid logLevel of '\(upperCaselogLevelString)'. '\(upperCaseVariable)' will NOT be set."))
+                }
+            }
+        }
+        return errors
+    }
+    
+    ///
+    /// Determine the LogLevel for this tag based on the configuration
+    ///
+    func logLevel(for tag: String) -> LogLevel {
+        
+        /// Set to the default global level first
+        var level = self.globalLogLevel
+        
+        /// Determine if there is a class log level first
+        if let tagLogLevel = self.loggedTags[tag] {
+            level = tagLogLevel
+        } else {
+            
+            /// Determine the prefixes log level if available
+            for (prefix, logLevel) in self.loggedPrefixes {
+                if tag.hasPrefix(prefix) {
+                    level = logLevel
```

Comment:
Yes, both would be good optimizations given the expected usage.  Will make the change.